tianshou/data/buffer/her.py

```diff
@@ -0,0 +1,180 @@
+from typing import Any, Callable, List, Optional, Tuple, Union
+
+import numpy as np
+
+from tianshou.data import Batch, ReplayBuffer
+
+
+class HERReplayBuffer(ReplayBuffer):
+    """Implementation of Hindsight Experience Replay. arXiv:1707.01495.
+
+    Currently support only 'future' strategy of HER.
+
+    :param int size: the size of the replay buffer.
+    :param compute_reward_fn: a function that takes 3 arguments: \
+        `acheived_goal`, `desired_goal`, `info` and returns the reward(s).
+        Note that the goal arguments can have extra batch_size dimension and in that \
+        case, the rewards of size batch_size should be returned
```

Comment:
updated: https://github.com/thu-ml/tianshou/pull/753/commits/63e65c087fa4c070861f12ba4d5fea130fc56740

Reply:
Here I also remove the third argument, `info`. Previously this argument was manually set to empty dict `{}` during the sampling because its shape was complicated. Originally I have it there because `Fetch` env's [`compute_reward`](https://github.com/Farama-Foundation/Gymnasium-Robotics/blob/main/gymnasium_robotics/envs/fetch_env.py#L67) function also takes `info` argument, although it was never used.